buyer-api/src/queues/transactionQueue.js

```diff
@@ -34,25 +37,33 @@ const createTransactionQueue = () => {
 
     switch (transaction.status) {
       case 'success': {
-        logger.info(`[tx][${name}] Transaction success ${receipt}`);
+        logger.info(`Tx[${id}] :: ${name} :: Success ${receipt}`);
         break;
       }
       case 'failure': {
-        logger.info(`[tx][${name}] Transaction failure ${receipt}`);
+        logger.info(`Tx[${id}] :: ${name} :: Failure ${receipt}`);
         break;
       }
       case 'pending': {
-        logger.info(`[tx][${name}] Transaction pending ${receipt}. Proceeding to retry...`);
+        logger.info(`Tx[${id}] :: ${name} :: Pending ${receipt} (will be retried)`);
         throw new Error('Retry tx');
       }
       default: {
-        logger.info(`[tx][${name}] Unknown transaction status`);
+        logger.info(`Tx[${id}] :: ${name} :: Unknown ${receipt} (will NOT be retried)`);
       }
     }
 
     return transaction;
   });
 
+  queue.on('failed', ({
+    id, failedReason, data: { name },
+  }) => {
+    if (failedReason !== 'Retry tx') {
+      logger.error(`Tx[${id}] :: ${name} :: Error thrown: ${failedReason} (will be retried)`);
```

Comment:
So if `(failedReason !== 'Retry tx')` it will be retried? 🤔

Reply:
I'm throwing an error a few lines above with that message so the job can be retried, and I don't want another log about it.

Reply:
But isn't that **if** confusing? If the reason is NOT `Retry tx`, it will be retried. (Or at least that's what is being output with the `logger.error`

Reply:
Yes, it is confusing! But I prioritized the output over this complexity. 

Without this check, the output would be:
```
info: Tx[10] :: NewOrder:: Pending 0x123... (will be retried)
error: Tx[10] :: NewOrder :: Error thrown: Retry tx (will be retried)
```

With this check only the first line is logged. Error outputs are left to real errors.

The root cause of this weird check is that I didn't come up with an simple mechanism to retry the Job from within the processor, and this is because the Job is locked while being processed. The easiest way to retry it is to throw an error.

Reply:
Ok I'm convinced now 👏